cmd/logs.go

```diff
@@ -83,7 +83,17 @@ func (c LogsCmd) buildTailCmd(opts LogsOpts) []string {
 
 	var logsDir string
 
-	if opts.Agent {
+	if (opts.System && !opts.Agent) || opts.All {
+		// Globbing makes it very difficult to do fine-grained exclusions of file types. Because we do not want to tail compressed files,
+		// rotated files, or the files in the `sysstat` directory (some of which are binary), we're using find to get fine-grained
+		// control.
+		// If folks complain that we're tailing something that makes their terminal sad, feel free to add additional filtering here.
+		// Also note that this string will eventually get executed as "sudo bash -c '$TAIL_COMMAND'", so we need to backslash-escape
+		// globbing characters, rather than wrapping them in single-quotes.
+		tail = append(tail, "$(find /var/log -type f -not -name \\*.gz -and -not -name \\*.xz -and -not -name \\*.\\[1-9] -and -not -path /var/log/sysstat/\\* -and -not -wholename /var/log/wtmp -and -not -wholename /var/log/lastlog)")
```

Comment:
This might be easier to read if a back-tick is used instead of double quotes so that double-escaping is not needed.

Reply:
True, it might. However, neither I nor my pair at the time had any idea what the "`" construction was (command execution, as in Ruby?? (_surely_ not, given the context)), and had to look it up when we encountered it. I do have a moderate-strength preference for keeping the amount of weird golang-specific knowledge required to understand a bit of code to a minimum.

Reply:
You could `scp` the code, or you could use the externalized `embed` file as a holding place for a string which you (eventually) pass to `bash -c ...`.

This wouldn't necessarily change the complexity of the command string that needs to be sent, though it might make it easier (as a single `.sh` file) to reason about.

The code might look something like:
```golang
//go:embed tail-command.sh
var tailCommand string

func (c LogsCmd) buildTailCmd(opts LogsOpts) []string {
    //....
    return []string{"sudo", "bash", "-c", "exec", tailCommand}
}
```

... the snipped above presupposes some way of signaling the code contained in `tail-command.sh` which behavior it should exhibit. This might be accomplished by env-vars, or by breaking up a monolithic file into separate chunks to be included/excluded the way the `buildTailCmd` currently works.

The thought being that if all the logic in the `buildTailCmd` function could be expressed in a bash script which could be passed to `bash -c exec ...` then the overall logic might be easier to comprehend.

Or possibly not 😬